metrics/metrics.go

```diff
@@ -172,6 +175,11 @@ func NewDefaultHandler(o Options) http.Handler {
 
 // NewMetrics creates a metrics collector instance based on the Format option.
 func NewMetrics(o Options) Metrics {
+	// custom metrics implementation has priority over Format configuration
+	if o.MetricsBackend != nil {
```

Comment:
this feels a bit like magic from the calling code's perspective. Since the calling case provides the options, it also just could us the MetricsBackend when available, without calling metrics.NewMetrics. It can be misleading also because in this case, all the other options are ignored.